nntrainer/layers/acti_func.h

```diff
@@ -54,6 +54,9 @@ class ActiFunc {
   template <typename T = float> void setActiFunc(ActivationType acti_type) {
     activation_type = acti_type;
 
+    if (typeid(T) == typeid(_FP16))
+      THROW_UNLESS_FP16_ENABLED;
```

Comment:
this needs to be checked again. I don't think we need to throw an error here. These activation function type surely need to support FP16 datatype. To check, we have to enable the 'enable-fp16' option in meson_options.txt should be true.

Reply:
This ensures thorwing an exception when FP16 activation is called without FP16 support.
By calling (conditional) THROW at the point where FP16 op actually exists, we can skip writing FP16 conditions at the callers (where you update codes mostly).

Reply:
This is an exception for a situation where attempting to use FP16 even though FP16 is disabled in the Meson option, so it seems proper to handle the exception here.